tools/build-version.go

```diff
@@ -19,29 +20,45 @@ func getTag(match ...string) (string, *semver.PRVersion) {
 	} else {
 		tagParts := strings.Split(string(tag), "-")
 		if len(tagParts) == 3 {
-			if ahead, err := semver.NewPRVersion(tagParts[1]); err == nil {
+			ahead, err := semver.NewPRVersion(tagParts[1])
+			if err == nil {
 				return tagParts[0], &ahead
 			}
+			log.Printf("semver.NewPRVersion(%s): %v", tagParts[1], err)
 		} else if len(tagParts) == 4 {
-			if ahead, err := semver.NewPRVersion(tagParts[2]); err == nil {
+			ahead, err := semver.NewPRVersion(tagParts[2])
+			if err == nil {
 				return tagParts[0] + "-" + tagParts[1], &ahead
 			}
+			log.Printf("semver.NewPRVersion(%s): %v", tagParts[2], err)
 		}
 
 		return string(tag), nil
 	}
 }
 
 func main() {
-	if tags, err := exec.Command("git", "tag").Output(); err != nil || len(tags) == 0 {
+	tags, err := exec.Command("git", "tag").Output()
+	if err != nil {
+		log.Println("git tag:", err)
+		if e, ok := err.(*exec.ExitError); ok {
+			log.Print(string(e.Stderr))
+		}
+	}
+	if len(tags) == 0 {
 		// no tags found -- fetch them
-		exec.Command("git", "fetch", "--tags").Run()
+		err = exec.Command("git", "fetch", "--tags").Run()
+		if err != nil {
+			log.Println("git fetch --tags:", err)
+		}
 	}
+
 	// Find the last vX.X.X Tag and get how many builds we are ahead of it.
 	versionStr, ahead := getTag("--match", "v*")
 	version, err := semver.ParseTolerant(versionStr)
 	if err != nil {
 		// no version tag found so just return what ever we can find.
+		log.Println("semver.ParseTolerant:", err)
```

Comment:
Print `versionStr` here too?

Reply:
Yes, definitely. I forget it yesterday.
Thank you for spotting!